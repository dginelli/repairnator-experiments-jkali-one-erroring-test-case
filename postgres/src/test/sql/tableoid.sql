BEGIN;

CREATE TABLE test_tableoid (
  id SERIAL8 NOT NULL PRIMARY KEY,
  title text
);

CREATE INDEX idxtest_tableoid ON test_tableoid USING zombodb(zdb(tableoid, ctid), zdb(test_tableoid)) WITH (url='http://localhost:9200/');

INSERT INTO test_tableoid (title) VALUES ('title1');
INSERT INTO test_tableoid (title) VALUES ('title2');
INSERT INTO test_tableoid (title) VALUES ('title3');

SET enable_seqscan TO off;
EXPLAIN (costs off) SELECT * FROM test_tableoid WHERE zdb(tableoid, ctid) ==> 'title*';

SET enable_seqscan TO on;
SET enable_indexscan TO off;
EXPLAIN (costs off) SELECT * FROM test_tableoid WHERE zdb(tableoid, ctid) ==> 'title*';

ABORT;