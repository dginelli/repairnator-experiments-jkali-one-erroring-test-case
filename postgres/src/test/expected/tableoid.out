BEGIN;
CREATE TABLE test_tableoid (
  id SERIAL8 NOT NULL PRIMARY KEY,
  title text
);
CREATE INDEX idxtest_tableoid ON test_tableoid USING zombodb(zdb(tableoid, ctid), zdb(test_tableoid)) WITH (url='http://localhost:9200/');
INSERT INTO test_tableoid (title) VALUES ('title1');
INSERT INTO test_tableoid (title) VALUES ('title2');
INSERT INTO test_tableoid (title) VALUES ('title3');
SET enable_seqscan TO off;
EXPLAIN (costs off) SELECT * FROM test_tableoid WHERE zdb(tableoid, ctid) ==> 'title*';
                             QUERY PLAN                             
--------------------------------------------------------------------
 Index Scan using idxtest_tableoid on test_tableoid
   Index Cond: (zdb((tableoid)::regclass, ctid) ==> 'title*'::text)
(2 rows)

SET enable_seqscan TO on;
SET enable_indexscan TO off;
EXPLAIN (costs off) SELECT * FROM test_tableoid WHERE zdb(tableoid, ctid) ==> 'title*';
                           QUERY PLAN                           
----------------------------------------------------------------
 Seq Scan on test_tableoid
   Filter: (zdb((tableoid)::regclass, ctid) ==> 'title*'::text)
(2 rows)

ABORT;
